engine/modelc/src/modelimporter_jni.cpp

```diff
@@ -979,7 +981,7 @@ JNIEXPORT jobject JNICALL Java_ModelImporter_LoadFromBufferInternal(JNIEnv* env,
         }
     }
 
-    if (!dmModelImporter::NeedsResolve(scene))
+    if (resolved && !dmModelImporter::NeedsResolve(scene))
```

Comment:
What will the behaviour be if `resolved` is `false` and we proceed to call `dmModelImporter::CreateJavaScene()`?

Reply:
In java, you will always have to call it at least once.
If all buffers were already resolved, then the user got a model and is good to go.

If the buffers aren't resolved, then the model is incomplete.

* From the editor side, we load the model once.
* If there are any missing buffers, we put those into a DataResolver map
* Then we call the model loader again, and this time with the data map so that it can find external buffer data.

The fix is  _only_ so that we only call Finalize exactly once.

The Finalize can be called either internally in modelimporter_gltf.cpp, if there was no need to load any extra buffers (all buffers were already resolved), and we can go ahead and return the result.
Or, it can be called here, if the jni has actually resolved all outstanding buffers.
And that only happends if there were outstanding buffers to begin with (i.e. the modelimporter_gltf.cpp didn't call Finalize)

Reply:
Ah, thank you for the explanation!